qa-tests-backend/src/test/groovy/NetworkFlowTest.groovy

```diff
@@ -698,57 +697,81 @@ class NetworkFlowTest extends BaseSpecification {
         }
 
         then:
-        "verify generated netpols vs current graph state"
+        "verify generated netpols against the current graph state"
         yamls.each {
             assert it."metadata"."namespace" != "kube-system" &&
                     it."metadata"."namespace" != "kube-public"
         }
         yamls.findAll {
             deployedNamespaces.contains(it."metadata"."namespace")
-        }.each {
-            String deploymentName =
-                    it."metadata"."name"["stackrox-generated-".length()..it."metadata"."name".length() - 1]
-            assert deploymentName != NOCONNECTIONSOURCE
-            assert it."metadata"."labels"."network-policy-generator.stackrox.io/generated"
-            assert it."metadata"."namespace"
-            def index = currentGraph.nodesList.findIndexOf { node -> node.deploymentName == deploymentName }
-            def allowAllIngress = deployments.find { it.name == deploymentName }?.createLoadBalancer ||
+        }.each { yaml ->
+            // Make sure that we have enough time to populate the graph: 30s for sensor + some extra.
+            // On the first run, this test benefits from the state left over by previous tests.
+            // However, if the first run fails, the env is recreated and the test is run immediately afterwards.
+            // The system requires time for the changes to propagate to sensor, thus we have an additional retry here.
+            Helpers.withRetry(6, 20) { retry ->
```

Comment:
🤔 I think by retrying here we might end up in the opposite situation than the original flake. When we call `generateNetworkPolicies` internally we create the network graph and generate the Network Policies based on it. If the assertions in this retry block fail it could be due to the graph being updated with new information since the call to `generateNetworkPolicies`. In other words, if we generate the Network Policies when the graph is not complete then this test will most likely fail even with the retries.

Reply:
I see your point and I also was thinking about this. The issue is that we are iterating over the yaml files generated for the network policy. If we regenerate the policies, then the retry would need to apply to the entire test. And actually it is the case currently! The only problem is that no each retry the deployments are teared down and we need to wait a warmup time before we can assert again. 

I could try to change this setup/teardown procedure to happen after the test has passed/failed after all retries. Would that do the job?

Reply:
My fear with changing the setup/teardown procedure is that I don't know how it could affect other tests in this class; we might be opening a pandora's box here. Maybe we can add another retry block in the `when` step and assert there `sourceDeploymentsFromNetworkPolicy.size() == 3` and `sourceDeploymentsFromNetworkPolicy.size() == 1`. WDYT?

Reply:
Looks great 🙂